wasm/leb128/leb128.go

```diff
@@ -94,26 +94,26 @@ func DecodeInt32(r io.Reader) (ret int32, num uint64, err error) {
 		}
 		ret |= (int32(b) & 0x7f) << shift
 		shift += 7
-		num++
+		bytesRead++
 		if b&0x80 == 0 {
 			if shift < 32 && (b&0x40) != 0 {
 				ret |= ^0 << shift
 			}
 			// Over flow checks.
 			// fixme: can be optimized.
-			if num > 5 {
+			if bytesRead > 5 {
 				return 0, 0, errOverflow32
-			} else if unused := b & 0b00110000; num == 5 && ret < 0 && unused != 0b00110000 {
+			} else if unused := b & 0b00110000; bytesRead == 5 && ret < 0 && unused != 0b00110000 {
 				return 0, 0, errOverflow32
-			} else if num == 5 && ret >= 0 && unused != 0x00 {
+			} else if bytesRead == 5 && ret >= 0 && unused != 0x00 {
 				return 0, 0, errOverflow32
 			}
 			return
 		}
 	}
 }
 
-func DecodeInt33AsInt64(r io.Reader) (ret int64, num uint64, err error) {
+func DecodeInt33AsInt64(r io.Reader) (ret int64, bytesRead uint64, err error) {
```

Comment:
Is there an int33 type?

Reply:
as in the latest spec: https://webassembly.github.io/spec/core/binary/instructions.html#control-instructions

> Unlike any other occurrence, the type index in a block type is encoded as a positive signed integer, so that its signed LEB128 bit pattern cannot collide with the encoding of value types or the special code 0x40, which correspond to the LEB128 encoding of negative integers. To avoid any loss in the range of allowed indices, it is treated as a 33 bit signed integer.

and this is used for decoding "block type" for control instructions which can be multiple post MVP, so I think that's why I implemented this and use in `ReadBlockType` function.  Even though this is overkilling for MVP spec, but I think this is harmless as it is

Reply:
thanks for giving me the contents for the comment 😎